File: leetcode/lc-us/cpp/2287.cpp
#include <bits/stdc++.h>

using namespace std;

class Solution {
public:
    int rearrangeCharacters(string s, string target) {
        int cnt1[123] = {}, cnt2[123] = {};
        for (char c: s) cnt1[c]++;
        for (char c: target) cnt2[c]++;
        int res = 1e8;
        for (char c: target) res = min(res, cnt1[c] / cnt2[c]);
        return res;
    }
};

File: leetcode/lc-us/py/2287.py
from collections import Counter


class Solution:
    def rearrangeCharacters(self, s: str, target: str) -> int:
        cnt1, cnt2 = Counter(s), Counter(target)
        res = float('inf')
        for key, value in cnt2.items():
            res = min(res, cnt1[key] // value)
        return res
